### PR TITLE
Sentry Token Validator spec

### PR DESCRIPTION
**Pull request by JoshVanL**
Adds documentation about sentry token validator spec, and control plane trust domain.

Closes https://github.com/dapr/docs/issues/4001

### CODE REVIEW

File: daprdocs/content/en/operations/security/mtls.md [hunk 1]
```diff
@@ -491,3 +491,52 @@ Watch this [video](https://www.youtube.com/watch?v=Hkcx9kBDrAc&feature=youtu.be&
 <div class="embed-responsive embed-responsive-16by9">
 <iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/Hkcx9kBDrAc?start=1400"></iframe>
 </div>
+
+### Sentry Token Validators
+
+Sentry can be configured to enable extra token validators beyond the Kubernetes bound Service Account validator, or replace the `insecure` validator enabled by default in self hosted mode.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/security/mtls.md]:
```suggestion
Sentry can be configured to:
- Enable extra token validators beyond the Kubernetes-bound Service Account validator, or
- Replace the `insecure` validator enabled by default in self-hosted mode
```

File: daprdocs/content/en/operations/security/mtls.md [hunk 2]
```diff
@@ -491,3 +491,52 @@ Watch this [video](https://www.youtube.com/watch?v=Hkcx9kBDrAc&feature=youtu.be&
...
+These are useful for joining extra non-Kubernetes clients to the Dapr cluster running in Kubernetes mode, or replacing the insecure "allow all" validator in self hosted mode to enable proper identity validation.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/security/mtls.md]:
```suggestion
You can join extra non-Kubernetes clients to the Dapr cluster running in Kubernetes mode, or replace the insecure "allow all" validator in self-hosted mode to enable proper identity validation.
```

File: daprdocs/content/en/operations/security/mtls.md [hunk 3]
```diff
@@ -491,3 +491,52 @@ Watch this [video](https://www.youtube.com/watch?v=Hkcx9kBDrAc&feature=youtu.be&
...
+The only token validator currently supported is the `jwks` validator.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/security/mtls.md]:
```suggestion
> The only token validator currently supported is the `jwks` validator.
```

File: daprdocs/content/en/operations/security/mtls.md [hunk 4]
```diff
@@ -491,3 +491,52 @@ Watch this [video](https://www.youtube.com/watch?v=Hkcx9kBDrAc&feature=youtu.be&
 <div class="embed-responsive embed-responsive-16by9">
 <iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/Hkcx9kBDrAc?start=1400"></iframe>
 </div>
+
+### Sentry Token Validators
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md]:
```suggestion
## Sentry service token validators
```

File: daprdocs/content/en/operations/security/mtls.md [hunk 5]
```diff
@@ -491,3 +491,52 @@ Watch this [video](https://www.youtube.com/watch?v=Hkcx9kBDrAc&feature=youtu.be&
...
+
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md:496]: Tokens are often used for authentication and authorization purposes. Token validators are components responsible for verifying the validity and authenticity of these tokens. For example in Kubernetes environments, a common approach to token validation is through the Kubernetes service account mechanism. This validator checks tokens against Kubernetes service accounts to ensure their legitimacy.

File: daprdocs/content/en/operations/security/mtls.md (hunk 1 above)
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md]:
```suggestion
Sentry service can be configured to enable extra token validators beyond the Kubernetes bound Service Account validator, or replace the `insecure` validator enabled by default in self hosted mode.
```

File: daprdocs/content/en/operations/security/mtls.md [hunk 6]
```diff
@@ -491,3 +491,52 @@ Watch this [video](https://www.youtube.com/watch?v=Hkcx9kBDrAc&feature=youtu.be&
...
+Sentry can be configured to enable extra token validators beyond the Kubernetes bound Service Account validator, or replace the `insecure` validator enabled by default in self hosted mode.
+These are useful for joining extra non-Kubernetes clients to the Dapr cluster running in Kubernetes mode, or replacing the insecure "allow all" validator in self hosted mode to enable proper identity validation.
+The only token validator currently supported is the `jwks` validator.
+
+### JWKS
+
+The `jwks` validator enables Sentry to validate JWT tokens using a JWKS endpoint.
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md]:
```suggestion
The `jwks` validator enables Sentry service to validate JWT tokens using a JWKS endpoint.
```

File: daprdocs/content/en/operations/security/mtls.md [hunk 7]
```diff
@@ -491,3 +491,52 @@ Watch this [video](https://www.youtube.com/watch?v=Hkcx9kBDrAc&feature=youtu.be&
...
+The contents of the token _must_ contain the `sub` claim which matches the SPIFFE identity of the Dapr client, in the same Dapr format `spiffe://<trust-domain>/ns/<namespace>/<app-id>`.
+The audience of the token must by the SPIFFE ID of the Sentry identity, e.g. `spiffe://cluster.local/ns/dapr-system/dapr-sentry`.
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md]:
```suggestion
The audience of the token must by the SPIFFE ID of the Sentry identity. For example, `spiffe://cluster.local/ns/dapr-system/dapr-sentry`.
```

File: daprdocs/content/en/operations/security/mtls.md [hunk 8]
```diff
@@ -491,3 +491,52 @@ Watch this [video](https://www.youtube.com/watch?v=Hkcx9kBDrAc&feature=youtu.be&
...
+Other basic JWT rules regarding signature, expiry etc. apply.
+
+The `jwks` validator can accept either a remote source to fetch the public key list or a static array for public keys.
+
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md:519]: Can you add a description for the example. Something like
"The configuration example below shows..."

File: daprdocs/content/en/operations/security/mtls.md [hunk 9]
```diff
@@ -491,3 +491,52 @@ Watch this [video](https://www.youtube.com/watch?v=Hkcx9kBDrAc&feature=youtu.be&
...
+    enabled: true
+    tokenValidators:
+      - name: jwks
+        options:
+          minRefreshInterval: 2m
+          requestTimeout: 1m
+          source: "https://localhost:1234/"
+          caCertificate: "<optional ca certificate bundle string>"
+```
+
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md:539]: Same here for a description of this example.

**Review by msfussell — COMMENTED** [review]
@JoshVanL Left some early comments. As always, if there is more you can add on the scenarios of when this would be useful, that helps tremendously. So add more of the why if possible.

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
small review on my part

**Review by msfussell — APPROVED** [review]
LGTM

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**